### DISCUSSION TITLE
Release v1.6.0

### DISCUSSION THREAD

**Discussion by aalmiray** [jreleaser/jreleaser#1367] [Announcements]
## Binaries

[https://github.com/jreleaser/jreleaser/wiki/Release-v1.6.0](https://github.com/jreleaser/jreleaser/wiki/Release-v1.6.0)

## 🚀 Features
**announce**
- e98ae98 Support OpenCollective as announcer, closes #1243

**catalog**
- 28021f1 Add cyclonedx as cataloger, closes #1250

**core**
- 33d2846 Add an active property to Artifact, Glob, FileSet, closes #1315

**deploy**
- 828a968 Check Nexus2 staging profile when run on dry-run mode, closes #1340
- 720f5ff Check if public key has been published, closes #1150
- bdba1f1 Support additional artifact types, closes #1135
- 04c6d60 Support NetBeans modules, closes #1135
- 1074285 Enable finer control when evaluating Maven Central rules, closes #1289 #1290

**hooks**
- 819f18e Let command & script hooks be conditionally executed, closes #1364
- 81de408 Support environment variables, closes #1351
- 40a4131 Support arbitrary scripts, closes #1344
- 90deefa Let command hooks specify their verbosity level, closes #1343
- fe47d2e Parse structure command output as environment properties, closes #1278

**jdks**
- 5036eb2 Configure unpack settings for disco/jdks mojos, closes #1366
- 813270f Support additional Disco API package parameters, closes #1365

**packagers**
- 635fb75 Add jib as packager, closes #989
- 0a5c4ba Support zip based distributions with Winget, closes #1253
- 91d737c Support SINGLE_JAR distribution in macports, closes #1257
- cb5fa28 Skip copying LICENSE file to tap/repository, closes #1255

**release**
- ea18959 Switch github-sdk implementation to Feign, closes #1302
- 429e8ad Add a branchPush property to releasers, closes #1296

## 🐛 Fixes
**announce**
- 165ebe5 Update Twitter4j to 4.1.2, closes #1361
- a64bc50 Configure changelog properties before resolving messages, closes #1307

**assemble**
- 20c43d4 Update Windows launcher settings, closes #1284
- 84bc80d Launchers should work without JAVA_HOME if java is found, closes #1284
- 4319b90 Use absolute paths when computing module names with jdeps, closes #1055
- 72d2295 Skip coying JARs if applications modular, closes #1267

**build**
- 2e5dc89 Fix math expression in update-website.sh, closes #1265

**changelog**
- 7c1f78d Conventional commit scope should allow more characters

**core**
- 16faf52 Validate flatpak.tap before setting tapName, closes #1323
- 2e9067d Resolve formulaName before cask validation, closes #1321
- 966a2e1 Fix checks in PlatformUtils, closes #1312
- d2fabaa Do not require Git remote for assemble, deploy, download
- 2e72f4d Update regex for CHRONVER, closes #1295
- 7ace55a Update formatted value for CycloneDX enum, closes #1263
- 484e4c2 Accidentally coalesced glob/regex pattern, closes #1270
- 5a98b23 Use absolute patterns as is when resolving globs, closes #1270

**deploy**
- 87fd4ea Handle pomchecker errors when run in dry-run mode, closes #1362
- a76aa02 Pomchecker 1.9.0+ updated its version banner

**gradle**
- db9a93f Add a flag to skip dependency on assembleDist task, closes #1316

**packagers**
- b97c7c1 Include all resolvable artifacts, closes #1329
- d6aecee Include missing .desktop file in flatpak manifests, closes #1330
- 1e86caa Identify all binary files for flatpak/spec, closes #1324
- 53d6168 Archive root entry name may not match archive name, closes #1324
- 11326ab prepare to use unversioned jdk dependency in brew, closes #1224
- 25d32f5 filter documentation & license URLs from flatpack, closes #1266

**release**
- a3ce74d Fix NPE when releasing on dry-run mode with bogus token, closes #1341
- 0950b0c Asset upload to GitHub via post, closes #1302

**signing**
- f8dcbab Do not fail signing if passphrase is null and dry-run is on, closes #1339
- c1a572a Fix cosign signature generation

**unscoped**
- 4ea3683 Show the reason when RestAPIException fails a build, closes #1357
- 600d7dd Add validation for tags to ChocolateyPackagerValidator, closes #1313

## 🔄️ Changes
**build**
- 7465e93 Update GH workflows with suggestions from actionlint

**changelog**
- b38d784 Fix log message for changelog.sort

**cli**
- c78b613 Simplify Banner code

**deploy**
- e84bcea Add timeouts to GPG key verifier

**gradle**
- f0eb1f0 Do not track task jreleaserInit due to file lock issues in Windows
- bc1a097 Reformat JReleaserInitTask

**packagers**
- 16d70ca Skip docker/jib publishing if dry-run mode is active, closes #989

**release**
- a528cb2 Deprecate codeberg, closes #1345
- aebbe4c Remove all references to hub4j, closes #1302

**sdks**
- 23f8dfc Remove zt-exec as dependency

**unscoped**
- 1a4ac3a Catalog depends on checksum
- e0c377e Fix Sonar violations
- 8a2e87a Fix Sonar violations
- ecee249 Reduce code duplication
- f4c9caa Update actions in templates
- 2b14797 Check command.exitValue in a case by case basis

## 🛠  Build

- becae31 Update reflect-config.json because of twitter4j
- 56035b2 Add OpenCollective token to release workflow
- 6648008 Deploy cyclonedx BOMs to Maven Central
- 67c771b Add a workflow to clear action caches
- bbbc2a3 Update smoketests settings
- ab8b9f2 fix dependencies required by native-image assembler
- 069b28a Print dependencies of jreleaser distribution before upload
- 07e06f1 Fix build breakage due to missing jreleaser-graalvm-java-sdk JAR
- a024778 Upgrade build to Gradle 7.6.1
- fb7bc6d Update to GraalVM 22.3.2
- d96180d Update release announcements
- 6239821 Simplify upload of provenance assets
- 4863191 Fix spotbugs violations
- 60d9936 Update kordamp plugins
- efe9fea Fix bugs found by Sonar
- 9c10890 Fix code quality issues found by Sonar
- 07bc045 Disable GraalVM Native Image assembler on Windows
- e97fbfa Fix build breakage
- 23f5dd3 Configure OpenCollective announcer
- 099cafa Update release announcements
- 1cba752 Use predictible order when collecting reflect-config data
- e974e8b Explicitly set java version/distro for codeQL GH workflow
- 9505173 Explicitly set java version/distro for build GH workflow, closes #1271
- 480ce83 Rollback wiki template update
- 17b1d3d Update wiki template

## 📝 Documentation

- 1806b64 Update release announcements [skip ci]
- ceea412 Add crschnick as a contributor for bug [skip ci]
- 9965efe Add ge0ffrey as a contributor for bug [skip ci]
- 2da0ac5 Add a “build from source” to readme [skip ci]
- 346e180 Add SimonVerhoeven as a contributor for code [skip ci]
- 516b68e Update badges [skip ci]
- e8d0f11 Add chiraqL as a contributor for code [skip ci]
- 529c81a Include code formatting section in contributing guidelines [skip ci]
- 330e52c Add klausbrunner as a contributor for code [skip ci]

## ⚙️  Dependencies

- 0ef94af update junit5 to 5.9.3
- 882c9b4 bump jfrog/frogbot from to 2.7.1
- afb47c2 update aws-java-sdk to 2.20.51
- 1261afa update maven-annotations to 3.8.2
- a5f0503 update jsonschema to 4.31.0
- 397b4d0 update picocli to 4.7.3
- 6cc2a29 update jsonschema to 4.30.0, closes #1213
- ca5ace0 bump jfrog/frogbot from to 2.6.5
- 849f2b5 bump jfrog/frogbot from to 2.6.4
- 67c892a Update openssf-scorecard to 2.1.3
- 4d9d1f1 rollback jacoco to 0.8.8
- 06a9c37 update picocli to 4.7.2
- c65bb4c Update feign to 12.3
- ab7901a update jacoco to 0.8.9
- 0417aae update plexus-archiver to 4.6.3
- 99d3015 Update pomchecker to 1.9.0
- 9f8e06a update failsafe to 3.3.1
- 491344b update greenmail to 2.0.0
- 7df95fb Update slf4j to 2.0.7
- 35f7add update json-smart to 2.4.10
- e0c0e71 update jsonschema to 4.29.0
- aebe67c update JavaEWAH to 1.2.3

---
- fccb6ee Releasing version 1.6.0
- 6496b42 Bump for next development cycle

## Contributors
We'd like to thank the following people for their contributions:
- Andres Almiray ([@aalmiray](https://github.com/aalmiray))
- Chirag Lamsal ([@chiraqL](https://github.com/chiraqL))
- Gauthier ([@gotson](https://github.com/gotson))
- Geoffrey De Smet
- Klaus Brunner
- Simon Verhoeven

<hr /><em>This discussion was created from the release <a href='https://github.com/jreleaser/jreleaser/releases/tag/v1.6.0'>Release v1.6.0</a>.</em>

**Comment by klausbrunner** [comment]
Wanted to try this out right away, but doing full-release on Github actions now gives me 

```
Error:  Failed to execute goal org.jreleaser:jreleaser-maven-plugin:1.6.0:full-release (default-cli) on project solarpos: Execution default-cli of goal org.jreleaser:jreleaser-maven-plugin:1.6.0:full-release failed: Unexpected error when creating release: java.io.IOException: Could not checkout branch main: Ref main cannot be resolved -> [Help 1]

Caused by: org.eclipse.jgit.api.errors.RefNotFoundException: Ref main cannot be resolved
    at org.eclipse.jgit.api.CheckoutCommand.call (CheckoutCommand.java:223)
    at org.jreleaser.sdk.git.GitSdk.checkoutBranch (GitSdk.java:262)
    at org.jreleaser.sdk.git.GitSdk.checkoutBranch (GitSdk.java:254)
    at org.jreleaser.sdk.github.GithubReleaser.createRelease (GithubReleaser.java:180)
    at org.jreleaser.sdk.git.release.AbstractReleaser.release (AbstractReleaser.java:59)
    at org.jreleaser.engine.release.Releasers.release (Releasers.java:56)
   ```

I initially suspected my PAT (set via env: JRELEASER_GITHUB_TOKEN) had expired, but that's still good and nothing else has changed except going from 1.5.1 to 1.6.0 really. I'll investigate further, but I'm wondering if there has been some breaking change?

**Reply by aalmiray** [reply]
Affected by a change introduced in https://github.com/jreleaser/jreleaser/issues/1296

It looks like _if_ branch `main` does not exist locally.

**Reply by aalmiray** [reply]
As far as I can tell:
- `named-auto-release.yml`: creates a commit on `main` branch and tags using Maven's release-plugin. OK. https://github.com/klausbrunner/solarpos/actions/runs/4839476242/jobs/8624520503#step:8:35
- `build-and-release.yml` is called with `git-ref` set to the recently created tag: `v0.3.2`. OK. https://github.com/klausbrunner/solarpos/actions/runs/4839476242/jobs/8624555685#step:1:41
- `HEAD` points to `b6c41b6`. OK. https://github.com/klausbrunner/solarpos/actions/runs/4839476242/jobs/8624555685#step:7:33

- [x] So far so good up to this point. The current branch should be set to `main` otherwise JReleaser would have printed out a warning that the configured branch differs from the detected one. Would have been great to capture the full `target/jreleaser/trace.log`, similarly as it's done [here](https://github.com/jreleaser/jreleaser/blob/baddc6be2472389b41ae8a3d1eb71ff6c9a263c0/.github/workflows/release.yml#L152-L159). Watch out for path names.

**Reply by klausbrunner** [reply]
Ah! Thanks for the hint. I had originally checked out the release tag like this:

```
      - name: Checkout
        uses: actions/checkout@v3
        with:
          fetch-depth: 0
          ref: v1.2.3
```
assuming that this would pull all the branches and history and _then_ git checkout the given ref (tag). However, it doesn't work like that, it's either-or. Changed to a two-step approach, and now it works as expected:

```
      - name: Checkout
        uses: actions/checkout@v3
        with:
          fetch-depth: 0
      - run: git checkout v1.2.3
```

**Reply by mthmulders** [reply]
This helped me out, too. Thanks for sharing it, @klausbrunner!

However, after using your two-step approach, JReleaser now fails when updating to the next release number (using `versions-maven-plugin:2.15.0:set` followed by a `git commit` and a `git push origin HEAD:main`:

```
[main 6d2d77f] chore: Prepare next version: 0.3.5-SNAPSHOT
 1 file changed, 1 insertion(+), 1 deletion(-)
To https://github.com/mthmulders/mcs
 ! [rejected]        HEAD -> main (non-fast-forward)
error: failed to push some refs to 'https://github.com/mthmulders/mcs'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.
```

Did you happen to run into something similar - and if so, how did you resolve it? Thanks in advance!
